### ISSUE TITLE
Bug: improper codegen for varags / wrong calling convention

### ISSUE THREAD

**Issue by MaulingMonkey** [microsoft/windows-rs#1592]
### Which crate is this about?

windows, windows-sys

### Crate version

0.33.0

### Summary

Several windows SDK APIs expect varags.  However, equivalent generated rust fns lack any `rest: ...` parameters.  Additionally, despite said APIs being explicitly `cdecl`, they are marked as `extern "system`" (`stdcall` on 32-bit windows) in the generated Rust code, which is the wrong calling convention.  The difference between call**er** and call**ee** stack cleanup should cause issues quickly.

# winsqlite3.h

```cpp
// C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\winsqlite\winsqlite3.h
SQLITE_API int SQLITE_CDECL sqlite3_config(int, ...);
SQLITE_API int SQLITE_CDECL sqlite3_db_config(sqlite3*, int op, ...);
SQLITE_API char *SQLITE_CDECL sqlite3_mprintf(const char*,...);
SQLITE_API char *SQLITE_CDECL sqlite3_snprintf(int,char*,const char*, ...);
SQLITE_API int SQLITE_CDECL sqlite3_test_control(int op, ...);
```
https://github.com/microsoft/windows-rs/blob/32c7eea14faabd083f5d3e6f0f03530a38ba2113/crates/libs/windows/src/Windows/Win32/System/SqlLite/mod.rs#L2957-L2959
https://github.com/microsoft/windows-rs/blob/32c7eea14faabd083f5d3e6f0f03530a38ba2113/crates/libs/sys/src/Windows/Win32/System/SqlLite/mod.rs#L135

# WinBer.h

```cpp
// C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\WinBer.h
WINBERAPI INT BERAPI ber_printf( BerElement *pBerElement, _In_ PSTR fmt, ... );
```
https://github.com/microsoft/windows-rs/blob/74350264eab09bcb62cafe00c611efcc52725ff8/crates/libs/windows/src/Windows/Win32/Networking/Ldap/mod.rs#L1240-L1245

# Rtutils.h

```cpp
// C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\Rtutils.h
DWORD
__cdecl
TracePrintfW(
    IN  DWORD       dwTraceID,
    IN  LPCWSTR     lpszFormat,
    IN  ...         OPTIONAL
    );

DWORD
__cdecl
TracePrintfExW(
    IN  DWORD       dwTraceID,
    IN  DWORD       dwFlags,
    IN  LPCWSTR     lpszFormat,
    IN  ...         OPTIONAL
    );
```

https://docs.rs/windows-sys/0.33.0/windows_sys/Win32/NetworkManagement/NetManagement/fn.TracePrintfW.html
https://docs.rs/windows-sys/0.33.0/windows_sys/Win32/NetworkManagement/NetManagement/fn.TracePrintfExW.html

**Comment by MaulingMonkey** [comment]
Ahh, found previous issues in `win32metadata` (was only searching this repository)
* <https://github.com/microsoft/win32metadata/issues/33>
* <https://github.com/microsoft/win32metadata/issues/800>

**Comment by kennykerr** [comment]
Yes, it looks like we could support such functions in Rust if the metadata actually supported them, which it does not at present.

**Comment by kennykerr** [comment]
Transferring this to the win32 metadata repo as there's nothing I can do to fix these functions until the metadata provides some hint about this.
